spec/features/user_contributes_new_councillor_spec.rb

```diff
@@ -18,6 +18,56 @@
       end
     end
 
+    scenario "on the contribution page" do
+      visit new_authority_councillor_contribution_path(authority.short_name_encoded)
+
+      expect(page).to have_content("Casey City Council")
+    end
+
+    context "when a person submit blank email" do
+      before :each do
+        visit new_authority_councillor_contribution_path(authority.short_name_encoded)
+
+        within ".councillor-contribution-councillors fieldset" do
+          fill_in "Full name", with: "Mila Gilic"
+          fill_in "Email", with: ""
+        end
+
+        click_button "Submit"
+      end
+
+      it "displays an error message" do
+        expect(page).to have_content("Email can't be blank")
+      end
+
+      it "does not go into the list of the suggested councillors" do
+        expect(page).to have_no_content("Full name: Mila Gilic")
+        expect(page).to have_no_content("Email:")
+      end
+    end
+
+    context "when a person submit an invalid email" do
```

Comment:
I really like the use of `context` and `before` here, makes it really easy to read. Nice one 👍